R-package/R/lgb.DataProcessor.R

```diff
@@ -0,0 +1,87 @@
+DataProcessor <- R6::R6Class(
+  classname = "lgb.DataProcessor",
+  public = list(
+    factor_levels = NULL,
+
+    process_y = function(y, objective, params) {
+
+      if (is.character(y)) {
+        y <- factor(y)
+      }
+
+      if (is.factor(y)) {
+
+        if (is.ordered(y)) {
+          warning("Passed labels as ordered factor. Order of levels is ignored.")
+        }
```

Comment:
After this PR, `lightgbm()` would determine the objective automatically according to the type of the `y` variable. Since `lightgbm` doesn't offer any ordinal regression objective, passing an ordinal variable would default to non-ordinal classification.

Reply:
Thanks for that explanation. Then since this is just informational, please remove this warning. I'd support adding a note *in the documentation* about ordering for factor-type labels not being important, but don't support the inclusion of a warning for something that isn't a problem requiring user action.

Reply:
Removed.

Reply:
thank you